cli/lockfile.rs

```diff
@@ -123,6 +197,83 @@ impl Lockfile {
     let checksum = crate::checksum::gen(&[code.as_bytes()]);
     self.content.remote.insert(specifier.to_string(), checksum);
   }
+
+  fn check_npm_package(
+    &mut self,
+    package: &NpmResolutionPackage,
+  ) -> Result<(), LockfileError> {
+    let specifier = package.id.serialize_for_lock_file();
+    if let Some(package_info) = self.content.npm.packages.get(&specifier) {
```

Comment:
Maybe we should error if we can't find the package? This would indicate that the user needs to reload with `--reload`. That said, maybe they would hit this error anyway downstream?

Reply:
Added an error and test